src/index.js

```diff
@@ -1,13 +1,28 @@
-const exec = require('exec-sync')
 const url = require('url')
 const Strategy = require('./strategy')
 
+function get_ports() {
+  //  Use execSync if it exist (Node 0.12 and later) or exec-sync
+  const execSync = require('child_process').execSync || require('exec-sync');
+  const ports = execSync('spurious ports --json');
+  if (typeof ports === 'string') {
+    return ports;
+  }
+
+  const Buffer = require('buffer').Buffer;
+  if (Buffer.isBuffer(ports)) {
+    return ports.toString();
+  }
+  
+  return "[error] Spurious service returned an unsupported result: " + ports;
```

Comment:
@JakeChampion I would actually prefer that but I tried to make it work as closely as they way it was before. I'll throw an error here and update the PR.